hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java

```diff
@@ -2590,17 +2590,27 @@ public OzoneFsServerDefaults getServerDefaults() throws IOException {
     long now = Time.monotonicNow();
     if ((serverDefaults == null) ||
         (now - serverDefaultsLastUpdate > serverDefaultsValidityPeriod)) {
-      serverDefaults = ozoneManagerClient.getServerDefaults();
-      serverDefaultsLastUpdate = now;
+      try {
+        for (ServiceInfo si : ozoneManagerClient.getServiceInfo()
+            .getServiceInfoList()) {
+          if (si.getServerDefaults() == null) {
+            continue;
+          }
+          serverDefaults = si.getServerDefaults();
+          serverDefaultsLastUpdate = now;
```

Comment:
nit: Should it exit the loop after first non-`null` server defaults found?

Reply:
Yes, I made the change. Thanks @adoroszlai